File: Week1/7 - Counting Elements/Solution.py
#
# Created on Fri Apr 10 2020
#
# Title: Leetcode - Counting Elements
#
# Author: Vatsal Mistry
# Web: mistryvatsal.github.io
#

class Solution:
    def countElements(self, arr):
        d = dict((ele, arr.count(ele)) for ele in arr)
        count = 0
        for ele in arr:
            if (ele + 1) in d.keys():
                count += 1
        return count

File: Week1/7 - Counting Elements/Solution.cpp
/*
 * Created on Fri Apr 10 2020
 *
 * Title: Leetcode - Counting Elements
 *
 * Author: Vatsal Mistry
 * Web: mistryvatsal.github.io
 */

#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    int countElements(vector<int>& arr) {
        unordered_map<int, int> umap;
        for(int i=0; i < arr.size(); i++) {
            umap[arr[i]]++;
        }
        int count = 0;
        for(int i=0; i < arr.size(); i++) {
            if(umap[arr[i] + 1] > 0) {
                count++;
            }
        }
        return count;
    }
};